src/main/java/pluto/command/DeleteCommand.java

```diff
@@ -0,0 +1,33 @@
+package pluto.command;
+
+import pluto.PlutoException;
+import pluto.Storage;
+import pluto.TaskList;
+import pluto.Ui;
+import pluto.task.Task;
+
+import java.io.IOException;
+
+public class DeleteCommand extends Command {
+    private int idx;
+
+    public DeleteCommand(int idx) {
+        this.idx = idx;
+    }
+
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws PlutoException {
+        Task t = tasks.deleteTask(idx);
+        try {
+            storage.rewriteFile(tasks);
+            StringBuilder deleteMessage = new StringBuilder();
+            deleteMessage.append("\tNoted. I've removed this task:\n");
+            deleteMessage.append(String.format("\t\t%s\n", t.toString()));
+            deleteMessage.append(String.format("\tNow you have %d tasks in the list.", tasks.nTasks()));
+            ui.print(deleteMessage);
+        } catch (IOException e) {
+            tasks.addTask(idx - 1, t);
+            throw new PlutoException("\tOOPS!!! Couldn't delete task. Try again!");
```

Comment:
I didn't like this solution for deleting a task from TaskList. 

Consider the scenario where your try block throws an Exception that is not IOException. Perhaps it throws an unchecked exception. In this scenario, you would lose the task to be added back into the TaskList because the catch block is not executed. This is a prime example of using Exceptions as control flow, which is something you should avoid.

Perhaps consider separating the file rewriting process from the task deletion process. The TaskList failing to be stored or rewritten should not mean that the delete command fails entirely.